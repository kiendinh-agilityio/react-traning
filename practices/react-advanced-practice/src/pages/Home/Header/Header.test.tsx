import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from '.';

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('Header Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Header currentPage="Dashboard" />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the current page title and matches snapshot', () => {
    const { getByText, asFragment } = render(
      <MemoryRouter>
        <Header currentPage="Dashboard" />
      </MemoryRouter>,
    );
    expect(getByText('Dashboard')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls navigate with /signin when logout button is clicked', () => {
    const { getByLabelText, asFragment } = render(
      <MemoryRouter>
        <Header currentPage="Dashboard" />
      </MemoryRouter>,
    );

    const logoutButton = getByLabelText('Button logout');
    fireEvent.click(logoutButton);

    expect(mockNavigate).toHaveBeenCalledTimes(1);
    expect(mockNavigate).toHaveBeenCalledWith('/signin');
    expect(asFragment()).toMatchSnapshot();
  });
});
