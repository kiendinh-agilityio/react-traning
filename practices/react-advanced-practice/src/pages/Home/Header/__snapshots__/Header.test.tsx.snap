// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component displays the current page title and matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex justify-between mb-7 px-[21px]"
  >
    <div
      class="rt-Box"
    >
      <div
        class="rt-Box"
      >
        <p
          class="rt-Text rt-r-weight-regular text-sm flex gap-[4px] text-base"
        >
          Pages / 
          <span
            class="rt-Box text-xs text-dark dark:text-light"
          >
            Tables
          </span>
        </p>
      </div>
      <p
        class="rt-Text rt-r-weight-regular text-sm font-bold text-dark mt-[6px] dark:text-light"
      >
        Dashboard
      </p>
    </div>
    <div
      class="rt-Flex rt-r-ai-center gap-[17px]"
    >
      <button
        aria-label="Button switch theme"
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl transparent cursor-pointer
   p-2 rounded-full bg-primary text-white"
        data-accent-color=""
      >
        <svg
          fill="none"
          height="32"
          viewBox="0 0 17 16"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_76_1062)"
          >
            <path
              d="M8.5 12C10.7091 12 12.5 10.2091 12.5 8C12.5 5.79086 10.7091 4 8.5 4C6.29086 4 4.5 5.79086 4.5 8C4.5 10.2091 6.29086 12 8.5 12Z"
              fill="#e52424"
            />
            <path
              d="M8.50001 2.39972C8.05782 2.39972 7.70001 2.0415 7.70001 1.59972V0.8C7.70001 0.358219 8.05782 0 8.50001 0C8.9422 0 9.30001 0.358219 9.30001 0.8V1.59972C9.30001 2.0415 8.9422 2.39972 8.50001 2.39972Z"
              fill="#e52424"
            />
            <path
              d="M13.0258 4.2743C12.8211 4.2743 12.6164 4.19618 12.4601 4.03993C12.1476 3.72752 12.1476 3.22099 12.4601 2.90858L13.0258 2.34305C13.3383 2.03074 13.8445 2.03065 14.157 2.34315C14.4695 2.65555 14.4695 3.16208 14.157 3.47449L13.5914 4.04002C13.4351 4.19618 13.2305 4.2743 13.0258 4.2743Z"
              fill="#e52424"
            />
            <path
              d="M15.7 8.8H14.9C14.4578 8.8 14.1 8.44178 14.1 8C14.1 7.55822 14.4578 7.2 14.9 7.2H15.7C16.1422 7.2 16.5 7.55822 16.5 8C16.5 8.44178 16.1422 8.8 15.7 8.8Z"
              fill="#e52424"
            />
            <path
              d="M13.5914 13.8912C13.3867 13.8912 13.182 13.8132 13.0258 13.6569L12.4601 13.0914C12.1476 12.779 12.1476 12.2725 12.4601 11.9601C12.7726 11.6476 13.2789 11.6475 13.5914 11.96L14.157 12.5255C14.4695 12.8379 14.4695 13.3444 14.157 13.6568C14.0008 13.8131 13.7961 13.8912 13.5914 13.8912Z"
              fill="#e52424"
            />
            <path
              d="M8.50001 16C8.05782 16 7.70001 15.6418 7.70001 15.2V14.4003C7.70001 13.9585 8.05782 13.6003 8.50001 13.6003C8.9422 13.6003 9.30001 13.9585 9.30001 14.4003V15.2C9.30001 15.6418 8.9422 16 8.50001 16Z"
              fill="#e52424"
            />
            <path
              d="M3.40858 13.8912C3.20389 13.8912 2.99921 13.8131 2.84296 13.6568C2.53046 13.3444 2.53046 12.8379 2.84296 12.5255L3.40858 11.96C3.72108 11.6477 4.22733 11.6476 4.53983 11.9601C4.85233 12.2725 4.85233 12.779 4.53983 13.0914L3.97421 13.6569C3.81796 13.8131 3.61327 13.8912 3.40858 13.8912Z"
              fill="#e52424"
            />
            <path
              d="M2.1 8.8H1.3C0.857813 8.8 0.5 8.44178 0.5 8C0.5 7.55822 0.857813 7.2 1.3 7.2H2.1C2.54219 7.2 2.9 7.55822 2.9 8C2.9 8.44178 2.54219 8.8 2.1 8.8Z"
              fill="#e52424"
            />
            <path
              d="M3.97421 4.27431C3.76952 4.27431 3.56483 4.19628 3.40858 4.04003L2.84296 3.4745C2.53046 3.16209 2.53046 2.65556 2.84296 2.34316C3.15546 2.03066 3.66171 2.03056 3.97421 2.34306L4.53983 2.90859C4.85233 3.221 4.85233 3.72753 4.53983 4.03994C4.38358 4.19619 4.17889 4.27431 3.97421 4.27431Z"
              fill="#e52424"
            />
          </g>
          <defs>
            <clippath
              id="clip0_76_1062"
            >
              <rect
                fill="white"
                height="16"
                transform="translate(0.5)"
                width="16"
              />
            </clippath>
          </defs>
        </svg>
      </button>
      <svg
        class="cursor-pointer"
        fill="none"
        height="12"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#a)"
          fill="#718096"
        >
          <path
            d="M7.647 2.135A2.211 2.211 0 0 0 6 1.435c-.648 0-1.234.247-1.65.696-.42.453-.626 1.07-.577 1.735C3.868 5.18 4.867 6.247 6 6.247c1.133 0 2.13-1.068 2.227-2.38.05-.66-.157-1.275-.58-1.732ZM9.781 11.06H2.22a.667.667 0 0 1-.52-.24.759.759 0 0 1-.154-.625c.181-1.006.747-1.851 1.636-2.444.79-.527 1.792-.817 2.82-.817 1.027 0 2.028.29 2.818.817.89.593 1.455 1.438 1.636 2.444a.759.759 0 0 1-.154.625.666.666 0 0 1-.52.24Z"
          />
        </g>
        <defs>
          <clippath
            id="a"
          >
            <path
              d="M.5.748h11v11H.5z"
              fill="#fff"
            />
          </clippath>
        </defs>
      </svg>
      <svg
        class="cursor-pointer"
        fill="none"
        height="13"
        width="13"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.5 4.625a1.875 1.875 0 1 0 0 3.75 1.875 1.875 0 0 0 0-3.75ZM10.548 6.5c0 .175-.014.35-.038.523l1.14.894a.272.272 0 0 1 .063.346l-1.08 1.864a.272.272 0 0 1-.331.115L8.96 9.704c-.28.216-.585.394-.91.532l-.2 1.424a.28.28 0 0 1-.27.231H5.42a.281.281 0 0 1-.27-.223l-.2-1.425a3.967 3.967 0 0 1-.91-.531l-1.342.539a.273.273 0 0 1-.33-.116l-1.08-1.863a.272.272 0 0 1 .061-.346L2.49 7.03a4.09 4.09 0 0 1-.038-.531c0-.175.014-.35.038-.524l-1.14-.893a.272.272 0 0 1-.062-.347l1.08-1.863a.273.273 0 0 1 .33-.116l1.342.54c.28-.216.585-.395.91-.532l.2-1.425a.28.28 0 0 1 .27-.23h2.16a.281.281 0 0 1 .27.223l.2 1.424c.326.136.632.315.911.532l1.34-.54a.273.273 0 0 1 .332.116l1.08 1.864a.272.272 0 0 1-.062.346l-1.141.894c.024.176.037.353.038.53Z"
          fill="#718096"
        />
      </svg>
      <svg
        class="cursor-pointer"
        fill="none"
        height="13"
        width="13"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m10.814 8.499-.114-.14c-.516-.623-.828-1-.828-2.765 0-.914-.218-1.664-.65-2.226-.317-.416-.747-.731-1.313-.964a.07.07 0 0 1-.02-.016C7.687 1.706 7.13 1.25 6.5 1.25s-1.186.457-1.39 1.139a.073.073 0 0 1-.018.015c-1.322.544-1.964 1.587-1.964 3.19 0 1.767-.311 2.143-.827 2.766l-.115.14a.824.824 0 0 0-.11.88.876.876 0 0 0 .806.495h7.24c.35 0 .655-.19.8-.492a.825.825 0 0 0-.108-.883ZM6.5 11.75a1.876 1.876 0 0 0 1.65-.988.094.094 0 0 0-.082-.137H4.932a.095.095 0 0 0-.08.045.093.093 0 0 0-.003.092 1.876 1.876 0 0 0 1.651.988Z"
          fill="#718096"
        />
      </svg>
      <button
        aria-label="Button logout"
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl transparent cursor-pointer
   text-gray"
        data-accent-color=""
      >
        <svg
          height="28"
          viewBox="0 0 256 256"
          width="28"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            style="stroke: none; stroke-width: 0; stroke-dasharray: none; stroke-linecap: butt; stroke-linejoin: miter; stroke-miterlimit: 10; fill: none; fill-rule: nonzero; opacity: 1;"
          >
            <path
              d="M69.313 54.442a2 2 0 0 1-1.634-3.149l10.118-14.399-10.118-14.399a2 2 0 0 1 3.273-2.3l10.926 15.549c.484.69.484 1.61 0 2.3L70.952 53.592a1.998 1.998 0 0 1-1.639.85zM57.693 30.092a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H9.759c-.013 0-.024.003-.037.004-.037 0-.074.008-.111.011-.122.009-.24.027-.354.057-.046.011-.089.026-.133.041A1.952 1.952 0 0 0 8.8.255c-.025.014-.053.019-.077.034-.012.007-.02.018-.031.025-.13.083-.252.177-.36.287-.019.019-.033.042-.051.061a2.088 2.088 0 0 0-.294.429 2.032 2.032 0 0 0-.129.316c-.013.042-.028.082-.038.125A1.955 1.955 0 0 0 7.759 2v69.787c0 .17.028.333.068.49.011.043.025.083.039.124.04.123.091.239.152.35.019.033.034.068.054.1.086.135.185.26.3.371.022.021.047.037.07.058.102.09.214.169.333.237.021.012.037.03.058.042l31.016 16.213a2.004 2.004 0 0 0 1.962-.061A2.002 2.002 0 0 0 42.775 88V73.787h14.918a2 2 0 0 0 2-2V45a2 2 0 0 0-4 0v24.787H42.775V18.213c0-.745-.414-1.428-1.074-1.772L17.902 4h37.791v24.092a2 2 0 0 0 2 2z"
              style="stroke: none; stroke-width: 1; stroke-dasharray: none; stroke-linecap: butt; stroke-linejoin: miter; stroke-miterlimit: 10; fill: #718096; fill-rule: nonzero; opacity: 1;"
              transform="matrix(2.81 0 0 2.81 1.407 1.407)"
            />
            <path
              d="M80.241 38.894H47.536a2 2 0 0 1 0-4h32.705a2 2 0 0 1 0 4z"
              style="stroke: none; stroke-width: 1; stroke-dasharray: none; stroke-linecap: butt; stroke-linejoin: miter; stroke-miterlimit: 10; fill: #718096; fill-rule: nonzero; opacity: 1;"
              transform="matrix(2.81 0 0 2.81 1.407 1.407)"
            />
          </g>
        </svg>
         Log Out
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Header Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex justify-between mb-7 px-[21px]"
  >
    <div
      class="rt-Box"
    >
      <div
        class="rt-Box"
      >
        <p
          class="rt-Text rt-r-weight-regular text-sm flex gap-[4px] text-base"
        >
          Pages / 
          <span
            class="rt-Box text-xs text-dark dark:text-light"
          >
            Tables
          </span>
        </p>
      </div>
      <p
        class="rt-Text rt-r-weight-regular text-sm font-bold text-dark mt-[6px] dark:text-light"
      >
        Dashboard
      </p>
    </div>
    <div
      class="rt-Flex rt-r-ai-center gap-[17px]"
    >
      <button
        aria-label="Button switch theme"
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl transparent cursor-pointer
   p-2 rounded-full bg-primary text-white"
        data-accent-color=""
      >
        <svg
          fill="none"
          height="32"
          viewBox="0 0 17 16"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_76_1062)"
          >
            <path
              d="M8.5 12C10.7091 12 12.5 10.2091 12.5 8C12.5 5.79086 10.7091 4 8.5 4C6.29086 4 4.5 5.79086 4.5 8C4.5 10.2091 6.29086 12 8.5 12Z"
              fill="#e52424"
            />
            <path
              d="M8.50001 2.39972C8.05782 2.39972 7.70001 2.0415 7.70001 1.59972V0.8C7.70001 0.358219 8.05782 0 8.50001 0C8.9422 0 9.30001 0.358219 9.30001 0.8V1.59972C9.30001 2.0415 8.9422 2.39972 8.50001 2.39972Z"
              fill="#e52424"
            />
            <path
              d="M13.0258 4.2743C12.8211 4.2743 12.6164 4.19618 12.4601 4.03993C12.1476 3.72752 12.1476 3.22099 12.4601 2.90858L13.0258 2.34305C13.3383 2.03074 13.8445 2.03065 14.157 2.34315C14.4695 2.65555 14.4695 3.16208 14.157 3.47449L13.5914 4.04002C13.4351 4.19618 13.2305 4.2743 13.0258 4.2743Z"
              fill="#e52424"
            />
            <path
              d="M15.7 8.8H14.9C14.4578 8.8 14.1 8.44178 14.1 8C14.1 7.55822 14.4578 7.2 14.9 7.2H15.7C16.1422 7.2 16.5 7.55822 16.5 8C16.5 8.44178 16.1422 8.8 15.7 8.8Z"
              fill="#e52424"
            />
            <path
              d="M13.5914 13.8912C13.3867 13.8912 13.182 13.8132 13.0258 13.6569L12.4601 13.0914C12.1476 12.779 12.1476 12.2725 12.4601 11.9601C12.7726 11.6476 13.2789 11.6475 13.5914 11.96L14.157 12.5255C14.4695 12.8379 14.4695 13.3444 14.157 13.6568C14.0008 13.8131 13.7961 13.8912 13.5914 13.8912Z"
              fill="#e52424"
            />
            <path
              d="M8.50001 16C8.05782 16 7.70001 15.6418 7.70001 15.2V14.4003C7.70001 13.9585 8.05782 13.6003 8.50001 13.6003C8.9422 13.6003 9.30001 13.9585 9.30001 14.4003V15.2C9.30001 15.6418 8.9422 16 8.50001 16Z"
              fill="#e52424"
            />
            <path
              d="M3.40858 13.8912C3.20389 13.8912 2.99921 13.8131 2.84296 13.6568C2.53046 13.3444 2.53046 12.8379 2.84296 12.5255L3.40858 11.96C3.72108 11.6477 4.22733 11.6476 4.53983 11.9601C4.85233 12.2725 4.85233 12.779 4.53983 13.0914L3.97421 13.6569C3.81796 13.8131 3.61327 13.8912 3.40858 13.8912Z"
              fill="#e52424"
            />
            <path
              d="M2.1 8.8H1.3C0.857813 8.8 0.5 8.44178 0.5 8C0.5 7.55822 0.857813 7.2 1.3 7.2H2.1C2.54219 7.2 2.9 7.55822 2.9 8C2.9 8.44178 2.54219 8.8 2.1 8.8Z"
              fill="#e52424"
            />
            <path
              d="M3.97421 4.27431C3.76952 4.27431 3.56483 4.19628 3.40858 4.04003L2.84296 3.4745C2.53046 3.16209 2.53046 2.65556 2.84296 2.34316C3.15546 2.03066 3.66171 2.03056 3.97421 2.34306L4.53983 2.90859C4.85233 3.221 4.85233 3.72753 4.53983 4.03994C4.38358 4.19619 4.17889 4.27431 3.97421 4.27431Z"
              fill="#e52424"
            />
          </g>
          <defs>
            <clippath
              id="clip0_76_1062"
            >
              <rect
                fill="white"
                height="16"
                transform="translate(0.5)"
                width="16"
              />
            </clippath>
          </defs>
        </svg>
      </button>
      <svg
        class="cursor-pointer"
        fill="none"
        height="12"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#a)"
          fill="#718096"
        >
          <path
            d="M7.647 2.135A2.211 2.211 0 0 0 6 1.435c-.648 0-1.234.247-1.65.696-.42.453-.626 1.07-.577 1.735C3.868 5.18 4.867 6.247 6 6.247c1.133 0 2.13-1.068 2.227-2.38.05-.66-.157-1.275-.58-1.732ZM9.781 11.06H2.22a.667.667 0 0 1-.52-.24.759.759 0 0 1-.154-.625c.181-1.006.747-1.851 1.636-2.444.79-.527 1.792-.817 2.82-.817 1.027 0 2.028.29 2.818.817.89.593 1.455 1.438 1.636 2.444a.759.759 0 0 1-.154.625.666.666 0 0 1-.52.24Z"
          />
        </g>
        <defs>
          <clippath
            id="a"
          >
            <path
              d="M.5.748h11v11H.5z"
              fill="#fff"
            />
          </clippath>
        </defs>
      </svg>
      <svg
        class="cursor-pointer"
        fill="none"
        height="13"
        width="13"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.5 4.625a1.875 1.875 0 1 0 0 3.75 1.875 1.875 0 0 0 0-3.75ZM10.548 6.5c0 .175-.014.35-.038.523l1.14.894a.272.272 0 0 1 .063.346l-1.08 1.864a.272.272 0 0 1-.331.115L8.96 9.704c-.28.216-.585.394-.91.532l-.2 1.424a.28.28 0 0 1-.27.231H5.42a.281.281 0 0 1-.27-.223l-.2-1.425a3.967 3.967 0 0 1-.91-.531l-1.342.539a.273.273 0 0 1-.33-.116l-1.08-1.863a.272.272 0 0 1 .061-.346L2.49 7.03a4.09 4.09 0 0 1-.038-.531c0-.175.014-.35.038-.524l-1.14-.893a.272.272 0 0 1-.062-.347l1.08-1.863a.273.273 0 0 1 .33-.116l1.342.54c.28-.216.585-.395.91-.532l.2-1.425a.28.28 0 0 1 .27-.23h2.16a.281.281 0 0 1 .27.223l.2 1.424c.326.136.632.315.911.532l1.34-.54a.273.273 0 0 1 .332.116l1.08 1.864a.272.272 0 0 1-.062.346l-1.141.894c.024.176.037.353.038.53Z"
          fill="#718096"
        />
      </svg>
      <svg
        class="cursor-pointer"
        fill="none"
        height="13"
        width="13"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m10.814 8.499-.114-.14c-.516-.623-.828-1-.828-2.765 0-.914-.218-1.664-.65-2.226-.317-.416-.747-.731-1.313-.964a.07.07 0 0 1-.02-.016C7.687 1.706 7.13 1.25 6.5 1.25s-1.186.457-1.39 1.139a.073.073 0 0 1-.018.015c-1.322.544-1.964 1.587-1.964 3.19 0 1.767-.311 2.143-.827 2.766l-.115.14a.824.824 0 0 0-.11.88.876.876 0 0 0 .806.495h7.24c.35 0 .655-.19.8-.492a.825.825 0 0 0-.108-.883ZM6.5 11.75a1.876 1.876 0 0 0 1.65-.988.094.094 0 0 0-.082-.137H4.932a.095.095 0 0 0-.08.045.093.093 0 0 0-.003.092 1.876 1.876 0 0 0 1.651.988Z"
          fill="#718096"
        />
      </svg>
      <button
        aria-label="Button logout"
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button  
    px-[22px] py-2.5 font-bold rounded-xl transparent cursor-pointer
   text-gray"
        data-accent-color=""
      >
        <svg
          height="28"
          viewBox="0 0 256 256"
          width="28"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            style="stroke: none; stroke-width: 0; stroke-dasharray: none; stroke-linecap: butt; stroke-linejoin: miter; stroke-miterlimit: 10; fill: none; fill-rule: nonzero; opacity: 1;"
          >
            <path
              d="M69.313 54.442a2 2 0 0 1-1.634-3.149l10.118-14.399-10.118-14.399a2 2 0 0 1 3.273-2.3l10.926 15.549c.484.69.484 1.61 0 2.3L70.952 53.592a1.998 1.998 0 0 1-1.639.85zM57.693 30.092a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H9.759c-.013 0-.024.003-.037.004-.037 0-.074.008-.111.011-.122.009-.24.027-.354.057-.046.011-.089.026-.133.041A1.952 1.952 0 0 0 8.8.255c-.025.014-.053.019-.077.034-.012.007-.02.018-.031.025-.13.083-.252.177-.36.287-.019.019-.033.042-.051.061a2.088 2.088 0 0 0-.294.429 2.032 2.032 0 0 0-.129.316c-.013.042-.028.082-.038.125A1.955 1.955 0 0 0 7.759 2v69.787c0 .17.028.333.068.49.011.043.025.083.039.124.04.123.091.239.152.35.019.033.034.068.054.1.086.135.185.26.3.371.022.021.047.037.07.058.102.09.214.169.333.237.021.012.037.03.058.042l31.016 16.213a2.004 2.004 0 0 0 1.962-.061A2.002 2.002 0 0 0 42.775 88V73.787h14.918a2 2 0 0 0 2-2V45a2 2 0 0 0-4 0v24.787H42.775V18.213c0-.745-.414-1.428-1.074-1.772L17.902 4h37.791v24.092a2 2 0 0 0 2 2z"
              style="stroke: none; stroke-width: 1; stroke-dasharray: none; stroke-linecap: butt; stroke-linejoin: miter; stroke-miterlimit: 10; fill: #718096; fill-rule: nonzero; opacity: 1;"
              transform="matrix(2.81 0 0 2.81 1.407 1.407)"
            />
            <path
              d="M80.241 38.894H47.536a2 2 0 0 1 0-4h32.705a2 2 0 0 1 0 4z"
              style="stroke: none; stroke-width: 1; stroke-dasharray: none; stroke-linecap: butt; stroke-linejoin: miter; stroke-miterlimit: 10; fill: #718096; fill-rule: nonzero; opacity: 1;"
              transform="matrix(2.81 0 0 2.81 1.407 1.407)"
            />
          </g>
        </svg>
         Log Out
      </button>
    </div>
  </header>
</DocumentFragment>
`;
